cfg/decode_hook.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cfg
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/util"
+	"github.com/mitchellh/mapstructure"
+)
+
+func customTypeDecodeHook() mapstructure.DecodeHookFuncType {
+	return func(
+		f reflect.Type,
```

Comment:
can we call these reflect types src and target?

Reply:
This has been the convention for decode hooks. I can still change it if you want.